pkg/asset/installconfig/aws/validation.go

```diff
@@ -205,26 +196,69 @@ func validateSubnets(ctx context.Context, meta *Metadata, fldPath *field.Path, s
 }
 
 func validateMachinePool(ctx context.Context, meta *Metadata, fldPath *field.Path, platform *awstypes.Platform, pool *awstypes.MachinePool, req resourceRequirements, poolName string) field.ErrorList {
+	var err error
 	allErrs := field.ErrorList{}
+
+	// Pool's specific validation.
+	// Edge Compute Pool / AWS Local Zones:
+	// - is valid when installing in existing VPC; or
+	// - is valid in new VPC when Local Zone name is defined
+	if poolName == types.MachinePoolEdgeRoleName {
+		if len(platform.Subnets) > 0 {
+			edgeSubnets, err := meta.EdgeSubnets(ctx)
+			if err != nil {
+				errMsg := fmt.Sprintf("%s pool. %v", poolName, err.Error())
+				return append(allErrs, field.Invalid(field.NewPath("subnets"), platform.Subnets, errMsg))
+			}
+			if len(edgeSubnets) == 0 {
+				return append(allErrs, field.Required(fldPath, "invalid install config. There is no valid subnets for edge machine pool"))
+			}
+		} else {
+			if pool.Zones == nil {
+				return append(allErrs, field.Required(fldPath, "invalid install config. zone field is required for edge machine pool"))
+			}
+			if len(pool.Zones) == 0 {
+				return append(allErrs, field.Required(fldPath, "invalid install config. zone field cannot be empty for edge machine pool"))
+			}
+
+			reEdgeZoneNames := regexp.MustCompile(awstypes.EdgeZoneNamesPattern)
+			invalidEdgeZone := false
+			invalidList := []string{}
+			for _, zone := range pool.Zones {
+				if !reEdgeZoneNames.MatchString(zone) {
```

Comment:
I appreciate your adding validation for this, but I have a few questions: 

1. Is the regex pattern used here "official" from published AWS docs?
2. Are there any services through the AWS sdk that could be used to validate a provided zone?

If we can use the AWS sdk to validate a provided zone, I think that is a much safer route than having a static regex. The problem we have had before is that if the regex is incorrect, and gives a false negative i.e. our validation says a zone is invalid when it is valid, then there are no workarounds to unblock users until we ship a patch. So in general, I think it is better to rely solely on the SDK for these types of validations whenever possible.

In fact, I'm wondering if you are already validating the edge zones with this check:
```go
if len(edgeSubnets) == 0 {
				return append(allErrs, field.Required(fldPath, "invalid install config. There is no valid subnets for edge machine pool"))
```

Reply:
> Is the regex pattern used here "official" from published AWS docs?

Yes:
- for [Availability Zones](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-availability-zones): "The code for Availability Zone is its Region code followed by a letter identifier. For example, us-east-1a."
- for [Local Zones](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-local-zones): "The code for a Local Zone is its Region code followed by an identifier that indicates its physical location. For example, us-west-2-lax-1 in Los Angeles."
- for [Wavelength Zones](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-wavelength-zones): "The code for a Wavelength Zone is its Region code followed by an identifier that indicates the physical location. For example, us-east-1-wl1-bos-wlz-1 in Boston."

> Are there any services through the AWS sdk that could be used to validate a provided zone?

Yes, the API [`DescribeAvailabilityZones`](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeAvailabilityZones.html) provides the way to query the [`AvailabilityZone`](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_AvailabilityZone.html) structure which has the attribute `zoneType` - also used when checking the subnet IDs when installing in existing VPC (filling the `EdgeSubnets` slice).

> If we can use the AWS sdk to validate a provided zone, I think that is a much safer route than having a static regex.

ACK and agree! I will replace the validation to check the API - I was unsure of querying the API in this stage, but it is really much safer.

> In fact, I'm wondering if you are already validating the edge zones with this check

This check will be valid only when installing a cluster in an existing VPC, the EdgeSubnets will query the ZoneType from the subnet added in the install-config. In this case, we don't know the subnet as the installer will create it, we need to validate the zone using other alternatives (regex, or AWS API, etc).